### PR TITLE
EZP-31190 Document using the URLService - URLs filtered by Section

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://jira.ez.no/browse/EZP-31190
| Versions      | master, 2.5

### CODE REVIEW

File: docs/api/public_php_api_managing_repository.md [hunk 1]
```diff
@@ -213,3 +213,30 @@ $languageCreateStruct->languageCode = 'ger-DE';
 $languageCreateStruct->name = 'German';
 $this->languageService->createLanguage($languageCreateStruct);
 ```
+
+## URLs
+
+### URLService
+
+[`URLService`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/URLService.php)
+enables you to find, load and update URLs.
```
Comment (adamwojs) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
enables you to find, load and update external URLs used in ezrichtext and ezurl field types.
```

File: docs/api/public_php_api_managing_repository.md [hunk 2]
```diff
@@ -213,3 +213,30 @@ $languageCreateStruct->languageCode = 'ger-DE';
...
+- query limit, if value is `0`, search query will not return any search hits
+
+```php
+$URLQuery = $this->URLService->newURLQuery();
+$URLQuery->filter = 'section';
+$URLQuery->sortClauses = 'descending';
+$URLQuery->offset = '0';
+$URLQuery->limit = '25'
+
+$this->URLService->findUrls($query);
```
Comment (adamwojs) [inline docs/api/public_php_api_managing_repository.md]:
```php
use eZ\Publish\API\Repository\Values\URL\Query\Criterion;
use eZ\Publish\API\Repository\Values\URL\Query\SortClause; 

# ...

$urlQuery = $this->urlService->newURLQuery();
$urlQuery->filter = new Criterion\LogicalAnd([
    new Criterion\SectionIdentifier('media'),
    new Criterion\Validity(true),
]);
$urlQuery->sortClauses = [
   new SortClause\URL(SortClause::SORT_DESC)
];
$urlQuery->offset = 0;
$urlQuery->limit = 25;

$this->urlService->findUrls($query);
```
Please test this code snippet before commit!

File: docs/api/public_php_api_managing_repository.md [hunk 3]
```diff
@@ -213,3 +213,30 @@ $languageCreateStruct->languageCode = 'ger-DE';
 $languageCreateStruct->name = 'German';
 $this->languageService->createLanguage($languageCreateStruct);
 ```
+
+## URLs
+
+### URLService
+
+[`URLService`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/URLService.php)
+enables you to find, load and update external URLs used in ezrichtext and ezurl field types.
```
Comment (DominikaK) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
enables you to find, load and update external URLs used in RichText and URL Fields.
```

File: docs/api/public_php_api_managing_repository.md [hunk 4]
```diff
@@ -213,3 +213,30 @@ $languageCreateStruct->languageCode = 'ger-DE';
...
+
+To view a list of all URLs, use [`URLService::findUrls`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/URLService.php#L33)
+
+`URLService::findUrls` takes as argument a [`URLQuery`,](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/Values/URL/URLQuery.php)
+in which you need to specify:
+
+- query filter e.g. section
```
Comment (DominikaK) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
- query filter e.g. Section
```

File: docs/api/public_php_api_managing_repository.md [hunk 5]
```diff
@@ -213,3 +213,30 @@ $languageCreateStruct->languageCode = 'ger-DE';
...
+- sort clauses for URL queries
```
Comment (DominikaK) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
- Sort Clauses for URL queries
```

File: docs/api/public_php_api_managing_repository.md [hunk 6]
```diff
@@ -213,3 +213,30 @@ $languageCreateStruct->languageCode = 'ger-DE';
...
+- offset for search hits, used for paging the results
+- query limit, if value is `0`, search query will not return any search hits
```
Comment (DominikaK) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
- query limit. If value is `0`, search query will not return any search hits
```

**Pull request merged**